src/Traits/ReflectionFunctionLikeTrait.php

```diff
@@ -219,6 +219,12 @@ public function getShortName(): string
 
     public function getStartLine(): int
     {
+        // php parser detect same line for has attrGroups
+        // attrGroup is mostly on next line
+        if ($this->functionLikeNode->attrGroups !== []) {
+            return $this->functionLikeNode->getAttribute('startLine') + count($this->functionLikeNode->attrGroups);
```

Comment:
Will it work for multi-line attributes? Like this:
```php
#[Deprecated]
class FooBar
{
    #[ArrayShape([
        'token' => 'string',
        'code' => 'integer'
    ])]
    public $foo;
}
```

I guess, this applies to all relevant getStartLine() methods too. Not sure, but check should be done

Reply:
That's an issue with php-parser, that can't detect line for attribute, as always fall for the stmt that hold it, so attrGroup or attr line = stmt line.

Reply:
I will look if that possible to tweak

Reply:
The solution is to use original start line from native reflection itself when attribute exists, see https://github.com/goaop/parser-reflection/pull/130/commits/a06b0c77f9e4fbae37d516203042a3ee9f71284c